### PR TITLE
Refactor: refactoring password-encoding

### PR DESCRIPTION
**Pull request by eapg**
Refactoring oauth2 repository to work with base64 and bcrypt
Creating validate and encode functions for base64 and bcrypt
refactoring test for use new encoding and decoding function

### CODE REVIEW

File: src/main/java/com/ea/restaurant/passwordEncoder/manager/PasswordEncoderManager.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+public class PasswordEncoderManager {
+  public static PasswordEncoder createPasswordEncoder(String passwordEncoderType) {
+    if (PasswordEncoderType.BASE64.toString().equalsIgnoreCase(passwordEncoderType)) {
+      return new PasswordEncoder(new Base64Encoder());
+    } else if (PasswordEncoderType.BCRYPT.toString().equalsIgnoreCase(passwordEncoderType)) {
+      return new PasswordEncoder(new BcryptEncoder());
+    } else {
+      throw new PasswordEncoderException();
+    }
+  }
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/passwordEncoder/manager/PasswordEncoderManager.java]: You can change this by:
```
if (PasswordEncoderType.BASE64.toString().equalsIgnoreCase(passwordEncoderType)) {
  return new PasswordEncoder(new Base64Encoder());
}

if (PasswordEncoderType.BCRYPT.toString().equalsIgnoreCase(passwordEncoderType)) {
  return new PasswordEncoder(new BcryptEncoder());
}

throw new PasswordEncoderException();
```

File: src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java [hunk 2]
```diff
@@ -45,6 +48,7 @@ public Oauth2ServiceImpl(
     this.appClientScopeRepository = appClientScopeRepository;
     this.appAccessTokenRepository = appAccessTokenRepository;
     this.appRefreshTokenRepository = appRefreshTokenRepository;
+    this.passwordEncoder = PasswordEncoderManager.createPasswordEncoder(passwordEncoderType);
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java]: You don't need to do this every time you want to use the encoder. Same as in previous python implementation, add the `PasswordEncoder` instance to the DI container.

File: src/test/resources/application.properties [hunk 3]
```diff
@@ -12,3 +12,5 @@ oauth2.secret.key=thisIsTheSecretKeyForTest-Etl001
 spring.data.mongodb.host=localhost
 spring.data.mongodb.port=27017
 spring.data.mongodb.database=ea_restaurant_java_etl
+
+password.encoding.type=BASE64
```
Comment (reynoldmorel) [inline src/test/resources/application.properties]: maybe ``security.password.encoding.type` is more specific?

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/main/java/com/ea/restaurant/passwordEncoder/manager/PasswordEncoderManager.java (hunk 1 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/passwordEncoder/manager/PasswordEncoderManager.java]: Done

File: src/test/resources/application.properties (hunk 3 above)
Comment (eapg) [reply src/test/resources/application.properties]: Done.

File: src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java (hunk 2 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/service/impl/Oauth2ServiceImpl.java]: Done.

File: src/main/java/com/ea/restaurant/config/SecurityConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+public class SecurityConfiguration {
+
+  @Bean
+  public PasswordEncoder passwordEncoderManager(
+      @Value("${security.password.encoding.type}") String passwordEncoderType) {
```
Comment (reynoldmorel) [inline src/main/java/com/ea/restaurant/config/SecurityConfiguration.java]: you can move this outside of this function so anyone could take advantage of this property within this class if needed

**Review by reynoldmorel — APPROVED** [review]
Added a small nit, but LGTM!

File: src/main/java/com/ea/restaurant/config/SecurityConfiguration.java (hunk 4 above)
Comment (eapg) [reply src/main/java/com/ea/restaurant/config/SecurityConfiguration.java]: Done.

**Pull request merged**